nrf-hal-common/src/twim.rs

```diff
@@ -392,6 +397,185 @@ where
             },
         )
     }
+
+    fn write_part(&mut self, buffer: &[u8], final_operation: bool) -> Result<(), Error> {
+        unsafe { self.set_tx_buffer(buffer)? };
```

Comment:
No fence required here?

Reply:
I'm not entirely sure. Added some more fences to be on the safe side.